### DISCUSSION TITLE
Best practice to sync namespaces & tokens from master to slave

### DISCUSSION THREAD

**Discussion by taiheyp** [apache/kvrocks#1719] [Q&A] [answered]
Here's the situation:

I've set up kvrocks in sentinel mode - 1 master + 2 slaves, and added some namespaces & tokens on the master node. In case of master failover, I had to add ns & tokens on slave nodes as well, or it would lead to `invalid password` error (even though the keys had already been synced).

So, is there a "best practice" to sync ns & tokens instead of doing this manually?

**Comment by git-hulk** [comment]
Yes, maybe we can support syncing tokens between master and slaves. To see if other guys feel good about this. @apache/kvrocks-committers

**Comment by caipengbo** [comment]
Yes, we should support this.

**Comment by ShooterIT** [comment]
I think namespace also is a kind of data, so let's do it🥳

**Comment by xiaobiaozhao** [comment]
BTY, Can we rethink the design of namespace?

**Reply by git-hulk** [reply]
Yes, I will post the design proposal recently.

**Comment by git-hulk** [comment, marked as answer]
Thanks all for supporting this, please move to issue #1720 if you have any further comments.
